logger_test.go

```diff
@@ -150,8 +150,10 @@ func TestNewSystemLogHook(t *testing.T) {
 			`\d{2}:\d{2}:\d{2}` +
 			// high-precision separator
 			`.` +
-			// nano-seconds
-			`\d{9}` +
+			// nano-seconds. Note that the quantifier range is
+			// required because the time.RFC3339Nano format
+			// trunctates trailing zeros.
+			`\d{1,9}` +
```

Comment:
just checking - I presume '0' comes out as '0', and hence always at least 1 digit? :-)

Reply:
hehe, same question

Reply:
That's a good question. It isn't documented afaics but observationally there is always a leading zero, yes. I've also discovered something rather interesting by looking at the parsing code:

- https://github.com/golang/go/blob/master/src/time/format.go#L252..L269

`time.RFC3339Nano` truncates trailing zeros because it is defined like this:

```
RFC3339Nano = "2006-01-02T15:04:05.999999999Z07:00"
```

But, you can create a non-truncating format by switching the values after the decimal point to zeros!:

```
const RFC3339NanoNoTruncate = "2006-01-02T15:04:05.000000000Z07:00"
```

Kinda surprising someone didn't add that as a standard option given the documented sorting behaviour of `time.RFC3339Nano`.

/cc @markdryan.